gnocchi/storage/file.py

```diff
@@ -34,10 +37,16 @@
     cfg.StrOpt('file_basepath',
                default='/var/lib/gnocchi',
                help='Path used to store gnocchi data files.'),
+    cfg.IntOpt('file_subdir_len',
```

Comment:
I'm not sure that solves the problem you had in mind entirely. If a filesystem is limited to 65535 items per directory, then splitting on anything more than the 2 first bytes (`2^16`) could lead to a problem. So you have to split on those first 2 bytes of the UUID to be sure your root directory does not exceed 65k items.
Then, in each subdirectory, you'll get the same problem.

Sounds like to we should create a directory every 2 bytes. A UUID being 32 chars long, that means having a structure of 16 directories long for each metric.

wdyt?

Reply:
yeah, this is what I was about to do, should we just hardcode 2 and that is ? Or should I keep the configuration option and put 2 by default ?

Reply:
I think you can keep it and set 2 by default.